compose/service.py

```diff
@@ -395,49 +395,80 @@ def _containers_have_diverged(self, containers):
     def execute_convergence_plan(self,
                                  plan,
                                  do_build=True,
-                                 timeout=DEFAULT_TIMEOUT):
+                                 timeout=DEFAULT_TIMEOUT,
+                                 detached=False):
         (action, containers) = plan
 
         if action == 'create':
             container = self.create_container(do_build=do_build)
+
+            log_stream = None
+            if container.has_api_logs and not detached:
+                log_stream = container.attach(stdout=True, stderr=True, stream=True)
+
             self.start_container(container)
 
-            return [container]
+            return zip([container], [log_stream])
```

Comment:
It would be really awesome if we could keep the same return value. I think a lot of the complexity in this change is having to accommodate this difference.

Since our log_stream comes from a container, I think we could make it a method on Container.

```python

class Container(object):
    def __init__(self, ...):
       ...
       self._log_stream = None


    def init_log_stream(self):
        if not container.has_api_logs:
            return
        self._log_stream = self.attach(stdout=True, stderr=True, stream=True)

    def get_log_stream(self):
        return self._log_stream
```

It's unfortunate that it wouldn't be immutable , but Container isn't immutable as it is, so I guess that's not so bad.

In this function I think we'll be able to restore the returning of just a list of containers, and it can do:

````python
    if not detached:
        container.init_log_stream()
````